control/state.py

```diff
@@ -7,109 +7,179 @@
 #  Authors: anita.shekar@ibm.com, sandy.kaur@ibm.com
 #
 
+import json
+import time
+import threading
 import rados
 import logging
-from typing import Dict, Optional
+from typing import Dict
+from collections import defaultdict
 from abc import ABC, abstractmethod
-from .generated import gateway_pb2 as pb2
-from google.protobuf import json_format
 
 
 class GatewayState(ABC):
-    """Persists gateway NVMeoF target state."""
+    """Persists gateway NVMeoF target state.
 
-    @abstractmethod
-    def add_bdev(self, bdev_name: str, val: str):
-        pass
+    Class attributes:
+        X_PREFIX: Key prefix for key of type "X"
+    """
 
-    @abstractmethod
-    def remove_bdev(self, bdev_name: str):
+    BDEV_PREFIX = "bdev_"
+    NAMESPACE_PREFIX = "namespace_"
+    SUBSYSTEM_PREFIX = "subsystem_"
+    HOST_PREFIX = "host_"
+    LISTENER_PREFIX = "listener_"
+
+    def get_state(self) -> Dict[str, str]:
+        """Returns state specification."""
```

Comment:
"specification" is really vague and can mean a ton of different things:
```suggestion
        """Returns the state dictionary."""
```

Reply:
👍